### ISSUE TITLE
BUG: Mouse Move Event Error: Uncaught (in promise) DOMException

### ISSUE THREAD

**Issue by KalebLester** [birchill/10ten-ja-reader#1110]
This issue was discovered when on [MangaDex](https://mangadex.org). After hovering over a clickable item once, it is no longer clickable. I tested this on multiple browser and in isolated environments with only 10ten enabled. The issue occurs on Chromium browser and Safari; I could not recreate this issue in Firefox. I am unaware of any other sites this issue affects.

The issue occurs on the menubar and the "Popular New Titles" section on the main page (you have to move the mouse over the entire element). The issue does not occur in the settings page.

I've recorded the issue on Microsoft Edge and I took a screenshot of the console in Safari. The video showcases how the error is logged in the console after every element is hovered.

https://user-images.githubusercontent.com/30964279/225940282-6136e1bb-7663-44ca-8219-39803ff6ecaa.mov

---

![SCR-20230317-eou](https://user-images.githubusercontent.com/30964279/225940664-ce37e4dd-0cfd-4a1b-a8cb-982c760bc52a.png)

---

The full error from Chromium is:
`Uncaught (in promise) DOMException: Failed to execute 'setEnd' on 'Range': The offset 1 is larger than the node's length (0).
    at adjustForRangeBoundary (chrome-extension://pnmaklegiibbioifkmfkgpfnmdehdfan/10ten-ja-content.js:4634:22)
    at rawCaretPositionFromPoint (chrome-extension://pnmaklegiibbioifkmfkgpfnmdehdfan/10ten-ja-content.js:4587:21)
    at caretPositionFromPoint (chrome-extension://pnmaklegiibbioifkmfkgpfnmdehdfan/10ten-ja-content.js:4521:20)
    at getTextFromCoveringLink (chrome-extension://pnmaklegiibbioifkmfkgpfnmdehdfan/10ten-ja-content.js:4937:24)
    at getTextAtPoint (chrome-extension://pnmaklegiibbioifkmfkgpfnmdehdfan/10ten-ja-content.js:4469:24)
    at ContentHandler.tryToUpdatePopup (chrome-extension://pnmaklegiibbioifkmfkgpfnmdehdfan/10ten-ja-content.js:11213:29)
    at ContentHandler.onMouseMove (chrome-extension://pnmaklegiibbioifkmfkgpfnmdehdfan/10ten-ja-content.js:10465:19)`

---

Extension version: 1.13.5
Note: I'm using the Chromium version on Edge; however, I do not believe this to be significant as the issue is present on Safari too.

**Comment by birtles** [comment]
Thanks for the excellent bug report. Sorry for the delay in getting back to you. I've dug into it and I believe I found the problem. It's in a code path not used by Firefox (since Firefox has better APIs for text lookup) so it's not surprising that it doesn't reproduce in Firefox.

Also, a side note, there is also a version for Edge in the Edge extensions store. It's almost identical to the Chrome version except for a few parts where the address of Chrome-specific pages has been updated for Edge.
